src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclint/HtmlTag.java

```diff
@@ -362,53 +355,60 @@ public boolean accepts(HtmlTag t) {
 
     TH(BlockType.TABLE_ITEM, EndKind.OPTIONAL,
             EnumSet.of(Flag.ACCEPTS_BLOCK, Flag.ACCEPTS_INLINE),
-            attrs(AttrKind.ALL, COLSPAN, ROWSPAN, HEADERS, SCOPE, Attr.ABBR,
-                VALIGN),
-            attrs(AttrKind.HTML4, AXIS, ALIGN, CHAR, CHAROFF),
-            attrs(AttrKind.USE_CSS, WIDTH, BGCOLOR, HEIGHT, NOWRAP)),
+            attrs(AttrKind.OK, COLSPAN, ROWSPAN, HEADERS, SCOPE, Attr.ABBR),
+            attrs(AttrKind.UNSUPPORTED, WIDTH, BGCOLOR, HEIGHT, NOWRAP, AXIS, ALIGN, CHAR, CHAROFF),
+            attrs(AttrKind.OK, VALIGN)), // Removed after JDK-8255214 fixed.
 
     THEAD(BlockType.TABLE_ITEM, EndKind.REQUIRED,
-            attrs(AttrKind.ALL, VALIGN),
-            attrs(AttrKind.HTML4, ALIGN, CHAR, CHAROFF)) {
+            attrs(AttrKind.UNSUPPORTED, ALIGN, VALIGN, CHAR, CHAROFF)) {
         @Override
         public boolean accepts(HtmlTag t) {
             return (t == TR);
         }
     },
 
-    TIME(HtmlVersion.HTML5, BlockType.INLINE, EndKind.REQUIRED),
+    TIME(BlockType.INLINE, EndKind.REQUIRED),
 
     TITLE(BlockType.OTHER, EndKind.REQUIRED),
 
     TR(BlockType.TABLE_ITEM, EndKind.OPTIONAL,
-            attrs(AttrKind.ALL, VALIGN),
-            attrs(AttrKind.HTML4, ALIGN, CHAR, CHAROFF),
-            attrs(AttrKind.USE_CSS, BGCOLOR)) {
+            attrs(AttrKind.UNSUPPORTED, ALIGN, CHAR, CHAROFF, BGCOLOR),
+            attrs(AttrKind.OK, VALIGN)) { // Removed after JDK-8255215 fixed
         @Override
         public boolean accepts(HtmlTag t) {
             return (t == TH) || (t == TD);
         }
     },
 
-    TT(HtmlVersion.HTML4, BlockType.INLINE, EndKind.REQUIRED,
+    TT(ElemKind.UNSUPPORTED, BlockType.INLINE, EndKind.REQUIRED,
             EnumSet.of(Flag.EXPECT_CONTENT, Flag.NO_NEST)),
 
     U(BlockType.INLINE, EndKind.REQUIRED,
             EnumSet.of(Flag.EXPECT_CONTENT, Flag.NO_NEST)),
 
     UL(BlockType.BLOCK, EndKind.REQUIRED,
             EnumSet.of(Flag.EXPECT_CONTENT),
-            attrs(AttrKind.HTML4, COMPACT, TYPE)) {
+            attrs(AttrKind.UNSUPPORTED, COMPACT, TYPE)) {
         @Override
         public boolean accepts(HtmlTag t) {
             return (t == LI);
         }
     },
 
-    WBR(HtmlVersion.HTML5, BlockType.INLINE, EndKind.REQUIRED),
+    WBR(BlockType.INLINE, EndKind.REQUIRED),
 
     VAR(BlockType.INLINE, EndKind.REQUIRED);
 
+    /**
+     * Enum representing the supportability of HTML element.
+     */
+    public static enum ElemKind {
+        OK,
+        INVALID,
+        OBSOLETE,
+        UNSUPPORTED
+    }
```

Comment:
OK: valid
OBSOLETE: obsolete, deprecated, but still supported (valid)
UNSUPPORTED: ever supported but no longer supported (invalid)
INVALID: the rest of others (invalid)

UNSUPPORTED can be used if we would like to choose a friendly message instead of saying "unknown tag" only.
OBSOLETE is not used anywhere in this commit.  Although HTML5 has some obsolete features, [JDK-8215577](https://bugs.openjdk.java.net/browse/JDK-8215577) didn't define them as valid features if my understanding is correct.  So I chose not to allow obsolete features in order to avoid inconsistency.

Reply:
For both `ElemKind` and `AttrKind` there only seem to be two kinds:
* valid
* previously valid

For these two cases, `OK` is obviously reasonable for `valid`, but `OBSOLETE` seems a better fit than `UNSUPPORTED`, but you could also use `HTML4` or `OLD_HTML4` or something like that to indicate why we're keeping the name around for better messages.  Or, stay with `UNSUPPORTED` but add a doc comment explaining that it was previously supported but no longer supported

Reply:
Ok, I will use `HTML4`.